Feature/implementsFilter2MapView

cordhernandez opened the pull request:
#141

cordhernandez committed:
small UI Update

cordhernandez committed:
Organizes Files Alphabetically

cordhernandez committed:
Search Filter View Controller updates

+ adds customClass SearchFilterViewController
+ adds mapView outlet

cordhernandez committed:
adds Async and UserDefaults

cordhernandez committed:
adds mapView methods

+ func loadStore()
+ func prepareMapView()
+ func loadStoresInMapView()
+ func populateStoreAnnotations()

Adds Aroma Messages
Adds Network Loading Indicator Code

cordhernandez committed:
fixes indention in StoresMapViewController

cordhernandez committed:
adds Map View Delegate Code

cordhernandez committed:
adds Driving Directions Code for Search Filter Map View

cordhernandez committed:
adds methods to Load Stores when User Pans

cordhernandez committed:
fixes map pin bug

changes BlacknectarMapPin to BlackNectarMapPin

cordhernandez committed:
small UI Update

SirWellington commented on BlackNectar/BlackNectar/SearchFilterViewController.swift:
```diff
@@ -0,0 +1,343 @@
+//
+//  MapViewSearchFilter.swift
```
This comment should match the filename.

cordhernandez replied:
woops!
changed.

SirWellington commented on BlackNectar/BlackNectar/SearchFilterViewController.swift:
```diff
@@ -0,0 +1,343 @@
+import MapKit
+import UIKit
+
+class SearchFilterViewController: UITableViewController, MKMapViewDelegate, CLLocationManagerDelegate {
```
`SearchFilterViewController` seems a little confusing. How about just `Filter` or `Search` ViewController?

cordhernandez replied:
FilterViewController it is!

SirWellington commented on BlackNectar/BlackNectar/SearchFilterViewController.swift:
```diff
@@ -0,0 +1,343 @@
+    var showFarmersMarkets = true
+    var showGroceryStores = true
+    
+    typealias callBack = ([Store]) -> ()
```
This typealias should only have to be defined once throughout the project. It can then be used throughout.

cordhernandez replied:
ooohhh ok 
got it 
done

SirWellington commented on BlackNectar/BlackNectar/SearchFilterViewController.swift:
```diff
@@ -0,0 +1,343 @@
+    fileprivate let async: OperationQueue = {
+        
+        let operationQueue = OperationQueue()
+        operationQueue.maxConcurrentOperationCount = 10
```
You probably don't need 10 threads here.

SirWellington replied:
1 or 2 is enough

cordhernandez replied:
2 sounds good

SirWellington commented on BlackNectar/BlackNectar/SearchFilterViewController.swift:
```diff
@@ -0,0 +1,343 @@
+//MARK: Loading Stores into Map View
+extension SearchFilterViewController {
+    
+    func loadStoresInMapView(at coordinate: CLLocationCoordinate2D) {
```
Nice. Can probably be `fileprivate`

cordhernandez replied:
done

SirWellington commented on BlackNectar/BlackNectar/SearchFilterViewController.swift:
```diff
@@ -0,0 +1,343 @@
+        let annotation = annotation as? CustomAnnotation
+        let smallSquare = CGSize(width: 30, height: 30)
+        let button = UIButton(frame: CGRect(origin: CGPoint.zero, size: smallSquare))
+        let blackNectarPin = UIImage(named: "BlackNectarMapPin")
```
You can probably create the UIImage once in the class, and then reuse it. It's never really gonna change.

cordhernandez replied:
done

SirWellington commented on BlackNectar/BlackNectar/SearchFilterViewController.swift:
```diff
@@ -0,0 +1,343 @@
+        let blackNectarPin = UIImage(named: "BlackNectarMapPin")
+        let annotationView = MKAnnotationView(annotation: annotation, reuseIdentifier: annotation?.identifier)
+        
+        button.setBackgroundImage(UIImage(named: "carIcon"), for: .normal)
```
Same for this carIcon

SirWellington commented on BlackNectar/BlackNectar/SearchFilterViewController.swift:
```diff
@@ -0,0 +1,343 @@
+        
+        LOG.debug("User dragged Map Screen to: \(center)")
+        
+        self.loadStoresInMapView(at: center)
```
Nice!

cordhernandez replied:
👊🏾

SirWellington commented on BlackNectar/BlackNectar/SearchFilterViewController.swift:
```diff
@@ -0,0 +1,343 @@
+    
+}
+
+fileprivate extension MKMapView {
```
I'm seeing duplicate code here. Suggest this code be put somewhere so that it is only written once.

cordhernandez replied:
ahhh good eye 
got this fixed mate

SirWellington commented on BlackNectar/BlackNectar/SearchFilterViewController.swift:
```diff
@@ -0,0 +1,343 @@
+        
+        LOG.warn("There are no stores around the users location (Stores loading result is 0)")
+        AromaClient.beginMessage(withTitle: "No stores loading result is 0")
+            .addBody("There are no stores around the users location (Stores loading result is 0 : \(additionalMessage)")
```
Would be nice to have the location that was searched where no results were found.

cordhernandez replied:
that would be nice 
done and done 
Big Brother

SirWellington commented on BlackNectar/BlackNectar/SearchFilterViewController.swift:
```diff
@@ -0,0 +1,343 @@
+fileprivate extension SearchFilterViewController {
+    
+    
+    func startSpinningIndicator() {
```
You can make a separate class called `UIViewControllers+` and add these two functions to an extension there.

cordhernandez replied:
that's smart 
I'll do that 
done

SirWellington commented on BlackNectar/BlackNectar/SearchFilterViewController.swift:
```diff
@@ -0,0 +1,343 @@
+}
+
+
+
```
What's up with all the empty lines down here?


🕳 🕳  🕳

cordhernandez replied:
I zoned out 
fixed though

SirWellington reviewed (changes requested):
Can't wait to see this in action.

cordhernandez committed:
adjusts swift files

+ Changes SearchFilterViewController to FilterViewController
+ adds UIViewControllers+

cordhernandez committed:
small updates to UI

cordhernandez committed:
Refactors/Cleans Code for entire file

cordhernandez committed:
removes file private and Network Loading Indicator Code

cordhernandez committed:
Removes Network Loading Indicator Code and adjusts Aroma Messages

cordhernandez committed:
adds UIViewControllers+.swift

adds methods for Network Activity Indicator

cordhernandez committed:
refactors/cleans code

SirWellington commented on BlackNectar/BlackNectar/FilterViewController.swift:
```diff
@@ -0,0 +1,261 @@
+    
+    var stores: [Store] = []
+    
+    var selectedPin: MKPlacemark?
```
These variables can probably be `private`, or `fileprivate`

cordhernandez replied:
upgrade to fileprivate complete

SirWellington commented on BlackNectar/BlackNectar/FilterViewController.swift:
```diff
@@ -0,0 +1,261 @@
+        prepareMapView()
+        loadStores()
+        userLocationInfoForAroma()
+        
```
One too many empty lines.

cordhernandez replied:
just one

cordhernandez replied:
fixed

SirWellington commented on BlackNectar/BlackNectar/UIViewControllers+.swift:
```diff
@@ -0,0 +1,26 @@
+import UIKit
+
+extension UIViewController {
+    
```
Hell yes!!

SirWellington reviewed (approved):
Wayyy bettta

cordhernandez committed:
refactors/cleans code

cordhernandez commented:
merging this branch
